### ISSUE TITLE
Accordion Header: Synchronize heading levels

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#71776]
### Description

Within any given accordion, all heading levels should be the same. An accordion is a group of content where each item is in an equivalent position in the hierarchy, so all headings inside a given accordion group needs to have the same heading level.

Right now, it's possible to set each item with a separate heading level, which creates an incorrect hierarchy. 

### Step-by-step reproduction instructions

1. Add an accordion block with at least two items.
2. Assign the first heading as h2
3. Assign the second heading as h4

This will set each heading at a separate level; but all accordion title blocks within the accordion group should be at the same level.

### Environment info

- WordPress trunk
- Gutenberg 21.7.0 RC1

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by levinbaria** [comment]
We can do that by providing the setting of heading level to accordion group and removing the heading levels(toolbar) from the individual accordion header and I have worked on this approach.

If this is suitable approach?, will add a PR for this.

**Comment by t-hamano** [comment]
Thanks for the suggestion.

> We can do that by providing the setting of heading level to accordion group and removing the heading levels(toolbar) from the individual accordion header 

I think this is a good approach.

Another approach might be to not move the heading level setting, but just show the snackbar. Like this:

<img width="966" height="566" alt="Image" src="https://github.com/user-attachments/assets/c59382e2-ab24-4ffc-9aca-cc93b0a54257" />

I don't have a strong opinion on either approach.

**Comment by joedolson** [comment]
I'm inclined to move the heading setting. Snackbars aren't the best for accessibility, since they're so far removed from the location where the setting was changed. Low vision users might just not notice them. 

But it's also a low-priority notification, which is the appropriate kind of notification for something like a snackbar.

**Comment by t-hamano** [comment]
@joedolson Thanks for the feedback!

> We can do that by providing the setting of heading level to accordion group and removing the heading levels(toolbar) from the individual accordion header

@levinbaria Let's try this approach.

**Comment by t-hamano** [comment]
Fixed in #71895
